### PR TITLE
GOV-134 livness/readiness check

### PR DESCRIPTION
**Pull request by somanath21**
## Description

* Check dependency service health before pod initialization.
* Add a link to the design document or include the design bullet points related to this PR here.

 _(Ignore if these details are present on the associated JIRA ticket)_

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Design related bullet points or design document link related to this PR added in the description above. 

- [ ] Updated corresponding Postman Collection or Api documentation for the changes in this PR.

- [ ] Create/update unit or integration tests for verifying the changes made.

- [ ] Add required Swagger annotation and update API documentation with details of any API changes if applicable

- [ ] Followed the naming conventions as given in https://docs.google.com/document/d/1Q4vaMSzrTxxh9TS0RILuNkSkYCxotuYk1Xe0CMIkkCU/edit?usp=sharing

### CODE REVIEW

File: helm/ph-ee-engine/ams-paygops/templates/deployment.yaml [hunk 1]
```diff
@@ -19,6 +19,11 @@ spec:
 {{ toYaml .Values.deployment.annotations | indent 8 }}
 {{- end }}
     spec:
+      initContainers:
+      #During this Pod's initialization, check that zeebe-gateway service is up and running before starting this pod
+      - name: check-zeebe-gateway-ready
+        image: busybox:latest
+        command: ['sh', '-c','until nc -vz g2p-sandbox-zeebe-gateway 26500; do echo "Waiting for myconfig service"; sleep 2; done;']
```
Comment (fynmanoj) [inline helm/ph-ee-engine/ams-paygops/templates/deployment.yaml:26]: `g2p-sandbox-` is coming from the release name. if you hard code this word, it is going to be difficult to manage the chart. use {{Release.name}} if possible

**Review by fynmanoj — CHANGES_REQUESTED** [review]
ports and service names are dynamic in certain cases. so while applying in scripts it should not be hard-coded.

File: helm/ph-ee-engine/ams-paygops/templates/deployment.yaml (hunk 1 above)
Comment (somanath21) [reply helm/ph-ee-engine/ams-paygops/templates/deployment.yaml]: Done

**Comment by somanath21** [comment]
> ports and service names are dynamic in certain cases. so while applying in scripts it should not be hard-coded.

the zeebe-gateway is running at 26000 across the environment, if any such changes can be done form the deployment yaml for both {{release-name}}-zeebe-gateway:26500 and for the init container

**Review by fynmanoj — APPROVED** [review]

**Pull request merged**